File: go/insertionSort/insertionSort_test.go
package insertionSort
import (
	"testing"
	"reflect"
)

func TestCase1(t *testing.T) {
	test1 := []int{4,3,2,1}
	output := InsertionSort(test1)
	expected := []int{1,2,3,4}

	if !reflect.DeepEqual(output, expected) {
		t.Errorf("Sort Array is incorrect")
	}
}

File: python/insertionSort_test_.py
import unittest
from insertionSort import insertionSort

class TestProgram(unittest.TestCase):
  def test_case1(self):
    test1=[4,3,2,1]
    output = insertionSort(test1)
    self.assertEqual(output ,[1,2,3,4])

if __name__ == "__main__":
  unittest.main()